src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs

```diff
@@ -70,5 +72,70 @@ public static string GetTransactionId(this Message message)
                 ? transactionId.ToString()
                 : string.Empty;
         }
+
+        /// <summary>
+        ///     Tries to read the Service Bus message body as a JSON structure, deserialized to the specified <typeparamref name="TMessage"/>.
+        /// </summary>
+        /// <typeparam name="TMessage">The type of the message to be deserialized.</typeparam>
+        /// <param name="message">The message to deserialize.</param>
+        /// <param name="output">The output to where the result should be written.</param>
+        /// <returns>
+        ///     Whether or not the deserialization succeeded [true], or not [false].
+        /// </returns>
+        public static bool TryReadBodyAsJson<TMessage>(this Message message, out TMessage output) where TMessage : class
```

Comment:
Maybe good to add some unit tests for these new methods? They are pretty important.

Reply:
Yeah, removed that. Leftover from legacy approach.